### DISCUSSION TITLE
Live ASR Engine

### DISCUSSION THREAD

**Discussion by Dimlight** [openai/whisper#508] [Q&A]
I have implemented (not from scratch) LiveASREngine using whisper using the following codebase: 

https://github.com/oliverguhr/wav2vec2-live

The only change I made was in the wav2vec2_inference.py: initialized whisper model with hugging face pipeline. 
my code: https://github.com/Dimlight/LiveASREngine 
The problem I am facing now: 

If I do not say anything and the entire room is silent, the engine continuously prints "you" or "thank you", I tested the system in a quiet room. still getting the same issue. 

Can anyone please help me what can be the reasons for getting this kind of result?

**Comment by ExtReMLapin** [comment]
It could be background noise but I had something similar.

I made a video with music, and at the very end of the video the music was fading away and the engine reported "Thank you for watching", this was too specific to just be a coincidence, I suspect OpenAI used a datased of video with subtitles, and some guy decided to add a subtitle "Thx for watching" even if no one was saying it.

**Reply by Dimlight** [reply]
I think I have spotted the problem of the code segments: 

We need to filter the sound waves somehow, in the current code it just passes whatever it gets: 
Code snippet from wav2vec2_inpference.py 

if len(frames) > 1:
                    asr_input_queue.put(frames)
frames = b''

need to add some filtering mechanism here.

**Reply by ExtReMLapin** [reply]
You could use a background noise removing model

**Reply by Dimlight** [reply]
Do you have any suggestions?

**Reply by ExtReMLapin** [reply]
nope
